src/components/materialui/ControlledAutocomplete.tsx

```diff
@@ -11,26 +12,46 @@ interface Props {
     onChangeCallback?,
     renderOption?: any,
     defaultValue?: any,
+    useSafeOptions?: boolean
 }
 
-const ControlledAutocomplete = ({options = [], name, renderInput, getOptionLabel, control, onChangeCallback, renderOption, defaultValue}: Props) => {
+const ControlledAutocomplete = ({options = [], name, renderInput, getOptionLabel, control, onChangeCallback, renderOption, defaultValue, useSafeOptions = false}: Props) => {
+    let _options = options;
+    let _defaultValue = defaultValue;
+    let _getOptionValue = (data) => data;
+
+    if (useSafeOptions) {
```

Comment:
To me this method is quite complex to understand mainly because you don't have may "const" here but many "let" ... which is harder to follow ... 
IMHO, it would help to split it into multiple functions and think about how to not use that many "let" expressions. (but I did not try to understand the function i detail ...)

Reply:
I will **const** make the code more readable.

Reply:
Will that make more sense?

```
    const prepareOptrions = () =>{
        let _options = options;
        let _defaultValue = defaultValue;
        let _getOptionValue = (data) => data;

        if (useSafeOptions) {
            const getKey = (o) => o?.iri ? o.iri : o?.uri ? o.uri : null
            const map: Map<string, any> = new Map()
            options.forEach(o => map.set(getKey(o), o))
            _defaultValue = defaultValue ? simplifyReferencesOfReferences(defaultValue) : null
            _options = options.map(o => simplifyReferencesOfReferences(o))
            _getOptionValue = (data) => {
                let key = getKey(data)
                key ? map.get(key) : data
            }
        }
        return [_options, _defaultValue, _getOptionValue]
    }
    
    const [_options, _defaultValue, _getOptionValue] = prepareOptrions()
```

Reply:
yes, looks better